src/generators/openapi/generators/openapi-schemas/generateSchemaFromModel.ts

```diff
@@ -0,0 +1,25 @@
+import { ApiBuilderModel } from 'apibuilder-js';
+import {
+  filter,
+  map,
+} from 'lodash';
+import {
+  generateSchemaPropertiesFromModelFields,
+} from '../openapi-schemas';
+
+function generateSchemaFromModel(model: ApiBuilderModel, modelValidator) {
+  const required = map(filter(model.fields, ['required', true]), req => req.name);
+  const properties = generateSchemaPropertiesFromModelFields(model.fields, modelValidator);
+
+  return {
+    [model.shortName]: {
+      description: model.description,
+      ...model.fields.length && { properties },
```

Comment:
This technique is new to me so forgive the nitpick.

This feels prone to error since if array is empty you are essentially doing `...0` and that doesn't feel right.

UPDATE: I guess so, tried it in an emulator.

Reply:
This was new to me also. I found the suggestion [here](https://medium.com/@mikeh91/conditionally-adding-keys-to-javascript-objects-using-spread-operators-and-short-circuit-evaluation-acf157488ede), but as you suggest arrays will be a slightly different case because of _falsy_ values (as pointed out in the comments of that article). Good catch.